shared-libs/contacts/test/unit/places.spec.js

```diff
@@ -336,27 +336,92 @@ describe('places controller', () => {
           type: 'person'
         }
       };
+
       sinon.stub(people, 'getOrCreatePerson').resolves({
         _id: 'qwe',
+        _rev: '1',
         name: 'Jim',
         type: 'person'
       });
-      fetchHydratedDoc.resolves({
-        _id: 'ad06d137',
-        name: 'CHP Branch One',
-        type: 'district_hospital'
+      db.medic.post.withArgs(
+        sinon.match((doc) => !doc.contact)
+      ).callsFake(doc => {
+        chai.expect(doc.name).to.equal('CHP Family');
+        chai.expect(doc.parent._id).to.equal('ad06d137');
+        return Promise.resolve({ id: 'hc', rev: '1' });
       });
-      db.medic.post.callsFake(doc => {
+      db.medic.post.withArgs(
+        sinon.match(doc => doc.contact)
+      ).callsFake(doc => {
+        chai.expect(doc.name).to.equal('CHP Family');
+        chai.expect(doc.parent._id).to.equal('ad06d137');
         chai.expect(doc.contact._id).to.equal('qwe');
         chai.expect(doc.contact.name).to.equal(undefined); // minified
-        chai.expect(doc.contact.type).to.equal(undefined); // minified
-        return Promise.resolve({id: 'ghi'});
+        chai.expect(doc.contact.type).to.equal(undefined); //
+        return Promise.resolve({ id: 'hc', rev: '2' });
       });
+
+      fetchHydratedDoc.withArgs('hc').resolves({
+        name: 'CHP Family',
+        type: 'health_center',
+        parent: {
+          _id: 'ad06d137',
+          name: 'CHP Branch One',
+          type: 'district_hospital'
+        }
+      });
+      fetchHydratedDoc.withArgs('ad06d137').resolves({
+        _id: 'ad06d137',
+        name: 'CHP Branch One',
+        type: 'district_hospital'
+      });
+
       return controller._createPlaces(place).then(actual => {
-        chai.expect(actual).to.deep.equal({id: 'ghi'});
+        chai.expect(db.medic.post.callCount).to.equal(2);
+        chai.expect(actual).to.deep.equal({
+          id: 'hc',
+          rev: '2',
+          contact: {
+            id: 'qwe',
+          }
+        });
+      });
+    });
+
+    it('returns err if contact does not have name', done => {
```

Comment:
Thanks, @freddieptf, for adding these unit tests. 
Here, can you please add another scenario similar to this one, to cover the error when the contact does not have the correct format or is missing attributes.

Reply:
The value of this test is really only seeing what happens when [contact validation](https://github.com/medic/cht-core/blob/b092815de12f6deb91b7d8ba789925f2ca346ac5/shared-libs/contacts/src/people.js#L32) fails when creating a place, i think adding more would be duplicating already [existing](https://github.com/medic/cht-core/blob/b092815de12f6deb91b7d8ba789925f2ca346ac5/shared-libs/contacts/test/unit/people.spec.js) tests